tests/devices/test_block.py

```diff
@@ -81,6 +82,13 @@ def test_block_signal_monitors_correct_pv(rw_rbv_block):
     assert rw_rbv_block.setpoint_readback.source.endswith("UNITTEST:MOCK:CS:SB:float_block:SP:RBV")
 
 
+def test_block_rw_with_weird_sp_sets_sp_suffix_correctly():
+    weird_sp_suffix = ":SP123"
+    block = BlockRw(float, MOCK_PREFIX, "block", sp_suffix=weird_sp_suffix)
+    assert block.readback.source.endswith("UNITTEST:MOCK:CS:SB:block")
+    assert block.setpoint.source.endswith("UNITTEST:MOCK:CS:SB:block:SP123")
```

Comment:
Can we have a test like this, but actually using the new `block_w` utility function and asserting that `setpoint.source == readback.source`? Since that's the "logic" which the new function does...